discord/commands/options.py

```diff
@@ -158,9 +159,6 @@ def __init__(self, input_type: Any = str, /, description: Optional[str] = None,
                         for i in input_type:
                             if i.__name__ == "GuildChannel":
                                 continue
-                            if isinstance(i, ThreadOption):
-                                self.channel_types.append(i._type)
-                                continue
```

Comment:
Following the comment I made above, this shouldn't be removed

Reply:
Was removed as part of the patch done for ThreadOptions. There is probably a better way around this however I couldn't come up with one. Currently, removing "ThreadOption" from channel_type_map and keeping these lines in gives me the following: `AttributeError: type object 'ThreadOption' has no attribute '_type'`. This is due to the `i` variable not actually being initialized and not having that attribute yet. Any ideas?

Reply:
What code are you using to reproduce this?

Reply:
For context, dev is a SlashCommandGroup and this is written inside a cog.
```py
    @dev.command()
    async def test_thread_join(self,
                               ctx: discord.ApplicationContext,
                               thread: discord.Option(
                                   discord.ThreadOption,
                                   description="Enter the thread for the bot to join.",
                                   required=True,
                               ),
                               ) -> None:
        """Testing thread joining."""
        await thread.join()
        await ctx.respond(f"Joined {thread.name}!")
```

Reply:
> Edit: Think I just realized what's wrong, should've passed `discord.ThreadOption("public")` instead. Will go ahead and revert those changes, however, shouldn't there be at least some documentation for `discord.ThreadOption`? I can't find any on the docs.
Agreed, it should be documented